packages/p2p/src/stores/advertiser-page-store.js

```diff
@@ -147,6 +152,37 @@ export default class AdvertiserPageStore extends BaseStore {
         this.setIsLoading(false);
     }
 
+    /** This function gets the advert info from the counterparty when user wants to view a shared advert */
+    getAdvertInfo() {
```

Comment:
Can we create a hook for this?

Reply:
hmmm we could, perhaps if we return a function on the action needed to take, either showing the modal or showing the edge cases? would that work?